test/RevokableTokenLock.spec.ts

```diff
@@ -0,0 +1,113 @@
+    });
+    it('should update value of revoker in contract to other address', async () => {
+      await revokableTokenLock.setRevoker(other.address);
+      const revoker = await revokableTokenLock.revoker();
+      expect(revoker).to.equal(other.address);
+    });
+  });
+  describe('#revoke', async () => {
+    const amount = ethers.utils.parseEther('20');
+    let transferred: BigNumber;
+    let remaining: BigNumber;
+    let unlockBegin: BigNumber;
+    let unlockCliff: BigNumber;
+    let unlockEnd: BigNumber;
+    let claimedAmounts: BigNumber;
+    let lockedAmounts: BigNumber;
+    before('Set up data', async () => {
+      ({unlockBegin, unlockCliff, unlockEnd} = await revokableTokenLock.vesting(recipient.address));
+      transferred = amount.mul(unlockCliff.sub(unlockBegin)).div(unlockEnd.sub(unlockBegin));
+      remaining = amount.sub(transferred);
+    });
+    it('should revert if caller is not owner or revoker', async () => {
+      await expect(revokableTokenLock.connect(other).revoke(recipient.address)).to.be.revertedWith(
+        'RevokableTokenLock: onlyAuthorizedActors'
+      );
+    });
+    it('should not revert if caller is revoker', async () => {
+      await expect(revokableTokenLock.connect(revoker).revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should not revert if caller is owner', async () => {
+      await expect(revokableTokenLock.revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should claim any tokens vested and revoke the rest when revoke is called', async () => {
+      // TODO: how to move this into a fixture?
+      await token.approve(revokableTokenLock.address, amount);
+      await revokableTokenLock.lock(recipient.address, amount);
+      await setNextBlockTimestamp(unlockCliff.toNumber());
+
+      // Call revoke and check token balances
+      await expect(() => revokableTokenLock.revoke(recipient.address)).to.changeTokenBalances(
+        token,
+        [recipient, owner],
+        [transferred, remaining]
+      );
+
+      // Check contract values
+      ({unlockEnd, claimedAmounts, lockedAmounts} = await revokableTokenLock.vesting(recipient.address));
+      expect(claimedAmounts).to.equal(transferred);
+      expect(lockedAmounts).to.equal(claimedAmounts);
+      expect(unlockEnd).to.equal(unlockCliff);
+    });
+    it('should emit the Revoked event when revoke is called', async () => {
+      // TODO: how to move this into a fixture?
+      await token.approve(revokableTokenLock.address, amount);
+      await revokableTokenLock.lock(recipient.address, amount);
+      await setNextBlockTimestamp(unlockCliff.toNumber());
+
+      // Call revoke
+      expect(await revokableTokenLock.revoke(recipient.address))
+        .to.emit(revokableTokenLock, 'Revoked')
+        .withArgs(recipient.address, remaining);
+    });
+    // TODO: mock revert on transfers to test transfer fails
+  });
```

Comment:
Can we also add another test that revoking _before_ the cliff returns zero and reclaims everything?

Reply:
Maybe also one that tries to revoke after the entire vesting schedule is over, and check that it does not lead to any weird resets where the user could claim again.

Reply:
done